rdbms/src/main/java/org/spine3/server/storage/jdbc/util/Closeables.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.storage.jdbc.util;
+
+import org.spine3.server.storage.jdbc.exception.MultipleCloseException;
+
+import java.util.Collection;
+import java.util.LinkedList;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A utility class for working with {@link AutoCloseable} instances.
+ *
+ * @author Dmytro Dashenkov.
+ */
+public class Closeables {
+
+    private Closeables() {
+    }
+
+    /**
+     * Closes all of the passed {@linkplain AutoCloseable AutoCloseables}.
+     *
+     * <p>In case of an {@linkplain Exception} on the {@link AutoCloseable#close()} tries to close
+     * all the rest instances.
+     *
+     * <p>The very first {@linkplain Exception} caught will be propagated in an {@link IllegalStateException}.
+     *
+     * @param closeables instances to close
+     * @throws IllegalStateException if {@linkplain AutoCloseable#close() close()} throws an {@link Exception}
+     */
+    public static void closeAll(Iterable<? extends AutoCloseable> closeables) {
+        checkNotNull(closeables);
+        final Collection<Exception> exceptions = new LinkedList<>();
+        for (AutoCloseable closable : closeables) {
+            try {
+                closable.close();
+            } catch (Exception e) {
+                exceptions.add(e);
+            }
+        }
+        if (exceptions.isEmpty()) {
+            return;
+        }
+        if (exceptions.size() == 1) {
+            final Exception cause = exceptions.iterator()
+                                              .next();
+            throw new IllegalStateException(cause);
+        }
+        throw new MultipleCloseException(exceptions);
```

Comment:
In this case you'd need to catch for two exceptions, which is not convenient.

You need to throw ISE, but put different cause because you'll look into the cause anyway.